internal/testing/require/exec.go

```diff
@@ -0,0 +1,66 @@
+/*
+ Copyright 2021 - 2022 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package require
+
+import (
+	"os/exec"
+	"sync"
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+// ShellCheck returns the path to the "shellcheck" executable or calls t.Skip.
+func ShellCheck(t testing.TB) string { t.Helper(); return shellcheck(t) }
+
+var shellcheck = executable("shellcheck", "--version")
+
+// executable builds a function that returns the full path to name.
+// The function (1) locates name or calls t.Skip, (2) runs that with args,
+// (3) calls t.Log with the output, and (4) calls t.Fatal if it exits non-zero.
+func executable(name string, args ...string) func(testing.TB) string {
+	var result func(testing.TB) string
+	var once sync.Once
+
+	return func(t testing.TB) string {
+		t.Helper()
+		once.Do(func() {
+			path, err := exec.LookPath(name)
+			cmd := exec.Command(path, args...) // #nosec G204 -- args from init()
+
+			if err != nil {
+				result = func(t testing.TB) string {
+					t.Helper()
+					t.Skipf("requires %q executable", name)
+					return ""
+				}
+			} else if info, err := cmd.CombinedOutput(); err != nil {
+				result = func(t testing.TB) string {
+					t.Helper()
+					assert.NilError(t, err, "%q\n%s", cmd.Args, info)
```

Comment:
Could we call t.Fatal directly rather than `assert.NilError` on an error? (Not sure this would actually be preferable.)

Reply:
Certainly. I was deferring to `gotest` to inspect and format the error. I believe that amounts to:

```go
t.Fatalf("error is not nil: %s: %q\n%s", err.Error(), cmd.Args, info)
```

- https://github.com/gotestyourself/gotest.tools/blob/v3.1.0/internal/assert/assert.go#L51-L53
- https://github.com/gotestyourself/gotest.tools/blob/v3.1.0/internal/assert/assert.go#L114

Is that preferable?

Reply:
No, I think it’s fine, I was just curious why you chose to assert.NilError. I really only noticed because the code comment specifically calls out t.Fatal

Reply:
I agree, that isn't obvious. I'll add a comment here about it.

Reply:
Done in amended [`e4f3f71`](https://github.com/CrunchyData/postgres-operator/pull/2981/commits/e4f3f71f73622e0307cb166f9070856bbcb729f9).